.github/actions/dockerhub_login/readme.md

```diff
@@ -0,0 +1,111 @@
+        env:
+          # This environment variable must have exactly this name to be used by docker push to sign the image above
+          # as implied by the `--disable-content-trust=false`
+          DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE: ${{ secrets.DOCKERHUB_KEY_PASSWORD }}
+```
+
+If you are not familar the "environment" tag above is a way to restrict access to secrets in github actions to specific users.   Usually requires a
+user to click accept on a "review" in the github action's UI.  You may also choose to use "github repository secrets", which allows any user with write
+access to the repository to access the secrets.
+
+Notice that sign occurs after the image has been built (no need pull secrets if the build fails or expose the secrets to the docker build.)
+
+## Parameters ##
+
+### username ###
+
+Dockerhub account username.
+
+### password ###
+
+Dockerhub account password.
+
+### key_material ###
+
+User's docker hub account private signing key's base64 encoded file's contents as generated by docker command line in the folder:
+`~/.docker/trust/private/${key_name}.key`, after it's content is piped through the base64 command to encode it.
+
+### key_name ###
+
+The key file's name as found in the `~/.docker/trust/private/${key_name}.key` folder after running a signing command locally. The key name appear to be the public key corresponding to the file's private key material contents.
+
+### key_password ###
+
+The key's passphrase, which will be required to an environment variable at `docker push` time named `DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE` as
+seen in the `Push to dockerhub.` step in the example above.
+
+## Notes on dockerhub keys ##
+
+First, some terminology.   Dockerhub has org's and repos.   Similar in url position to github's org's and repos.   A dockerhub image name is usually of the form `<org>/<repo>:<tag>`.   It's entirely possible when working locally or with extremly popular images that the image may not have a repo.
+Example, debian, ubuntu, etc.
+
+Now on the keys.  There are several pub/private keys used to handle signing in Docker Notary, and the documention isn't targeted toward usage.
+
+So let's cover the keys.
+
+* First there is an **org key** held by a member of the Diem Association in our case.
+* Second there is a **repositiory key**, aka a key that can must be signed by the **org key** and can be used to allow users to sign docker hub images bound for one **repository** ie:  validator, tools, mint, etc.
+* Third the key used by this action will be neither of the above, it will be a third **user key** that must be signed by the holder of the
+     **repository keys** to allow the signing of the docker image to be considered valid by this **user key**.
+
+Finally, the user key's privilage to sign any repo can be revoked/granted at any time, and is information stored by the dockerhub notary service, not in the key file.
+
+## Example Docker Key Commands ##
+
```

Comment:
@coryhill I no longer have access to the slack channel where I gave you instructions on how to do the key signing for dockerhub, if you could dig it up and comment/update this file once landed I'm sure future versions of us would appreciate it.